src/command.rs

```diff
@@ -216,8 +212,61 @@ impl Context {
         artifactory.publish(package, repository).await
     }
 
+    /// Helper method to traverse dependencies and install recursively.
+    #[async_recursion]
+    async fn traverse_and_install(
```

Comment:
Why is this on the ctx?

Reply:
Personal preference. I'm not a big fan of nesting functions inside other functions, I feel like it is not very readable or intuitive.

Reply:
I agree on the readability but i think they communicate an important message to the next engineer working on this.